patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -319,5 +325,77 @@ internal static class ModOrganizer
 				return new HashSet<string>();
 			}
 		}
+
+		private static readonly Regex PublishFolderMetadata = new Regex(@"[/|\\]([0-9]{4}[.][0-9]{1,2})[/|\\]");
```

Comment:
Here's an even stricter regex that won't match on invalid dates like `|2022.45|`

`[/|\\]([0-9]{4}[.](0?[1-9]|1[0-2]))[/|\\]`